File: backend/src/main/java/com/NBE3_4_2_Team4/standard/dto/PageDto.java
package com.NBE3_4_2_Team4.standard.dto;

import lombok.Getter;
import org.springframework.data.domain.Page;

import java.util.List;

@Getter
public class PageDto<T> {
    private final int currentPageNumber;
    private final int pageSize;
    private final long totalPages;
    private final long totalItems;
    private final boolean hasMore;
    private final List<T> items;

    public PageDto(Page<T> page) {
        this.currentPageNumber = page.getNumber() + 1;
        this.pageSize = page.getSize();
        this.totalPages = page.getTotalPages();
        this.totalItems = page.getTotalElements();
        this.hasMore = page.hasNext();
        this.items = page.getContent();
    }
}

File: backend/src/main/java/com/NBE3_4_2_Team4/standard/dto/PageDto.kt
package com.NBE3_4_2_Team4.standard.dto

import org.springframework.data.domain.Page

open class PageDto<T>(
    val currentPageNumber: Int,
    val pageSize: Int,
    val totalPages: Long,
    val totalItems: Long,
    val hasMore: Boolean,
    val items: List<T>
) {
    constructor(page: Page<T>) : this(
        currentPageNumber = page.number + 1,
        pageSize = page.size,
        totalPages = page.totalPages.toLong(),
        totalItems = page.totalElements,
        hasMore = page.hasNext(),
        items = page.content
    )
}
